tree_test.go

```diff
@@ -917,6 +931,39 @@ func TestInsertStem(t *testing.T) {
 	}
 }
 
+func TestInsertStemTouchingBothHalves(t *testing.T) {
```

Comment:
We never had a test for `InsertStem(...)` that would merge new values at both sides of the leaf node values vector, so [we had some case uncovered](https://github.com/gballet/go-verkle/blob/master/tree.go#L1060-L1063).